src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/show_app_menu_request.cc

```diff
@@ -0,0 +1,139 @@
+                         rpc_service,
+                         hmi_capabilities,
+                         policy_handler) {}
+
+ShowAppMenuRequest::~ShowAppMenuRequest() {}
+
+void ShowAppMenuRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "Application is not registered");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  if (mobile_apis::HMILevel::HMI_FULL != app->hmi_level() ||
+      helpers::Compare<mobile_apis::SystemContext::eType,
+                       helpers::NEQ,
+                       helpers::ALL>(
+          app->system_context(),
+          mobile_apis::SystemContext::SYSCTXT_MAIN,
+          mobile_apis::SystemContext::SYSCTXT_MENU)) {
+    LOG4CXX_ERROR(logger_, "Application is not activated");
+    SendResponse(false, mobile_apis::Result::REJECTED);
+    return;
+  }
+
+  smart_objects::SmartObject received_msg_params =
+      (*message_)[strings::msg_params];
+
+  smart_objects::SmartObject msg_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+  msg_params[strings::app_id] = app->app_id();
+
+  if (received_msg_params.keyExists(strings::menu_id)) {
+    const int32_t menu_id = received_msg_params[strings::menu_id].asInt();
+    if (!app->FindSubMenu(menu_id)) {
+      LOG4CXX_ERROR(logger_, "Menu with id " << menu_id << " is not found.");
+      SendResponse(false, mobile_apis::Result::INVALID_ID);
+      return;
+    }
+    msg_params[strings::menu_id] = menu_id;
+  }
+
+  StartAwaitForInterface(HmiInterfaces::HMI_INTERFACE_UI);
+  SendHMIRequest(hmi_apis::FunctionID::UI_ShowAppMenu, &msg_params, true);
+}
+
+void ShowAppMenuRequest::on_event(const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  switch (event.id()) {
+    case hmi_apis::FunctionID::UI_ShowAppMenu: {
+      EndAwaitForInterface(HmiInterfaces::HMI_INTERFACE_UI);
+      hmi_apis::Common_Result::eType result_code =
+          static_cast<hmi_apis::Common_Result::eType>(
+              message[strings::params][hmi_response::code].asInt());
+      std::string response_info;
+      GetInfo(message, response_info);
+      const bool result = PrepareResultForMobileResponse(
+          result_code, HmiInterfaces::HMI_INTERFACE_UI);
+
+      ApplicationSharedPtr application =
+          application_manager_.application(connection_key());
+
+      if (!application) {
+        LOG4CXX_ERROR(logger_, "NULL pointer");
+        return;
+      }
+
+      SendResponse(result,
+                   MessageHelper::HMIToMobileResult(result_code),
+                   response_info.empty() ? NULL : response_info.c_str(),
+                   &message[strings::msg_params]);
+      break;
+    }
+    default: {
+      LOG4CXX_ERROR(logger_, "Received unknown event" << event.id());
```

Comment:
@mkorniichuk minor thing, but i think it's worth inserting a whitespace here for better log readability

Reply:
@mked-luxoft fixed within ad229382f8df084c7a63027248807c7f7f03448d